src/lib.rs

```diff
@@ -42,456 +42,417 @@
+            return hex_to_ansi(hex.as_str(), templater);
+        }
+
+        // if it's not a formatter or hex code, it's a color code
+        let color = &captures["color"];
+
+        if self.depth == Depth::High {
             let [r, g, b] =
-                [0, 2, 4].map(|i| u8::from_str_radix(&code[i..i + 2], 16).unwrap().to_string());
+                COLORS_24BIT(color).expect("the regex should match only valid color codes");
 
-            Some(fill_rgb_template(formats[&24u8], &r, &g, &b))
-        } else if let Some(value) = FORMATTERS.get(code) {
-            Some(fill_template(formats[&3u8], value))
-        } else {
-            let template = formats[&(self.depth as u8)];
+            return fill_rgb_template(templater(Depth::High), r, g, b);
+        }
 
-            if self.depth == Depth::High {
-                let [r, g, b] = COLORS_24BIT.get(code)?;
+        let color_map =
+            colors(self.depth).expect("at this point depth should only be TTY, Low or Medium");
 
-                return Some(fill_rgb_template(template, r, g, b));
-            }
+        let mapped = color_map(color).expect("the regex should match only valid color codes");
 
-            let color_map = COLORS[&(self.depth as u8)];
-            let mut value = color_map.get(code)?.to_string();
+        // low bit depths use different way of specifying background
+        let value = if bg && self.depth <= Depth::Low {
+            (mapped
+                .parse::<u8>()
+                .expect("color tables should contain valid numbers")
+                + 10)
+                .to_string()
+        } else {
+            mapped.to_string()
+        };
 
-            if bg && self.depth <= Depth::Low {
-                value = (value.parse::<u8>().ok()? + 10).to_string()
-            };
+        fill_template(templater(self.depth), &value)
+    }
 
-            Some(fill_template(template, &value))
-        }
+    /// Writes the prompt to stdout, then reads a line from input,
+    /// and returns it (excluding the trailing newline).
+    pub fn input(&self, prompt: &str) -> std::io::Result<String> {
+        print!("{}", self.convert(prompt));
+        stdout().flush()?;
+
+        let mut inp = String::new();
+        stdin().read_line(&mut inp)?;
+        Ok(inp.trim_end().to_owned())
     }
 
     /// Resets the formatting back to the default.
-    pub fn reset(&self) {
-        print!("{}", self.convert(format!("{}r", self.marker)));
+    pub fn reset() {
+        print!("\x1b[0m");
+    }
+
+    /// Clears the current line
+    pub fn clear_line() {
+        print!("\x1b[2K");
+    }
+
+    /// Clears the screen
+    pub fn clear_screen() {
+        print!("\x1b[2J");
+    }
+
+    /// Escape all format markers in a string
+    ///
+    /// # Example
+    /// ```rust
+    /// # use dahlia::Dahlia;
+    /// let d = Dahlia::default();
+    /// let str = d.escape("&aHello &cWorld");
+    /// assert_eq!(str, "&_aHello &_cWorld");
+    /// ```
```

Comment:
Curious: is `# use dahlia::Dahlia;` some special syntax for the docs?

Reply:
Yes, it makes the line hidden in the preview.
https://doc.rust-lang.org/rustdoc/write-documentation/documentation-tests.html#hiding-portions-of-the-example

Reply:
That's cool